pkg/services/ngalert/migration/cond_trans.go

```diff
@@ -157,7 +154,17 @@ func transConditions(ctx context.Context, alert *migrationStore.DashAlert, store
 
 			rawFrom := newRefIDsToTimeRanges[refID][0]
 			rawTo := newRefIDsToTimeRanges[refID][1]
-			calculatedInterval, err := calculateInterval(legacydata.NewDataTimeRange(rawFrom, rawTo), simpleJson, nil)
+
+			// We check if the minInterval stored in the model is parseable. If it's not, we use "1s" instead.
+			// The reason for this is because of a bug in legacy alerting which allows arbitrary variables to be used
+			// as the min interval, even though those variables do not work and will cause the legacy alert
+			// to fail with `interval calculation failed: time: invalid duration`.
+			if _, err := interval.GetIntervalFrom(ds, simpleJson, time.Millisecond*1); err != nil {
```

Comment:
Is there a scenario where `interval` and `intervalMs` could be already set?

Reply:
From what I've seen in my tests, and if I can believe the comment above GetIntervalFrom(): https://github.com/grafana/grafana/blob/82f3127e23f2b5cad27c48e166fc781bc2369b6f/pkg/services/ngalert/migration/cond_trans.go#L343-L347

Then it `intervalMs` shouldn't be set in legacy alerting. That being said, I did go back and forth a bit on whether I should set the `interval` to an empty string instead of `1s`.